### PR TITLE
Add type safety with optional type parameters with @test<RequestBody, ResponseBody>

### PR DESCRIPTION
**Pull request by fwouts**

### CODE REVIEW

File: lib/src/test/examples/contract.ts [hunk 1]
```diff
@@ -76,7 +76,7 @@ class CreateUser {
     @body body: ErrorBody
   ) {}
 
-  @test({
+  @test<CreateUserRequestBody, UserBody>({
```
Comment (Kastet) [inline lib/src/test/examples/contract.ts:79]: What effect do these parameter have? Request and response body will be checked against these types? Will `allowInvalidRequest` option still work in this case?

Comment (lfportal) [reply lib/src/test/examples/contract.ts:79]: They will enforce type-safety at compile time - and also provide type hints for the IDE. `allowInvalidRequest` will not work as expected when the Request parameter is used.

Comment (Kastet) [reply lib/src/test/examples/contract.ts:79]: Ok, then we also need to update the doc https://github.com/airtasker/spot/wiki/Contract-Testing

Comment (fwouts) [reply lib/src/test/examples/contract.ts:79]: Before we go and update the docs @Kastet does the approach look OK to you?

Comment (Kastet) [reply lib/src/test/examples/contract.ts:79]: yeah it's a nice improvement!

**Review by Kastet — APPROVED** [review]

**Review by lfportal — APPROVED** [review]

File: lib/src/test/examples/contract.ts (hunk 1 above)
Comment (fwouts) [reply lib/src/test/examples/contract.ts:79]: PR is now merged. @Kastet what did you want to change in the wiki? I just noticed the examples don't actually have a request body at the moment.

**Pull request merged**